### DISCUSSION TITLE
Amasty_GdprCookie

### DISCUSSION THREAD

**Discussion by nul77** [breezefront/community#232] [3. Integrations]
Hi,

I'm attempting to make the module Amasty_GdprCookie compatible with the Breeze Evolution theme.
I've created a module and modified all javascript files according to this [https://breezefront.com/docs/integration-tips#reusing-luma-files](url)
All files are added to <referenceBlock name="breeze.js">
All the modified js files are loaded in my browser, but every file that returns a uiComponent has an error: "TypeError: Cannot read properties of undefined (reading 'extend')"

```js
define([
    'Amasty_GdprFrontendUi/js/modal-component',
    'jquery',
    'mage/translate',
    'Amasty_GdprFrontendUi/js/model/cookie-data-provider'
], function (
    Component,
    $,
    $t,
    cookieDataProvider,
) {
    'use strict';

    return Component.extend({
        component: 'Amasty_GdprFrontendUi/js/cookies',
        defaults: {
            template: 'Amasty_GdprFrontendUi/components/elems',
            allowLink: '/',
            firstShowProcess: '0',
            cookiesName: [],
            domainName: '',
            setupModalTitle: $t('Please select and accept your Cookies Group'),
            isPopup: false,
            isDeclineEnabled: false,
            barLocation: null,
            selectors: {
                barSelector: '[data-amcookie-js="bar"]',
                acceptButton: '[data-amgdprcookie-js="accept"]',
                closeCookieBarButton: '[data-amcookie-js="close-cookiebar"]'
            }
        },
        <methods removed for brevity>
    });
});
```

Any thoughts what I'm doing wrong?

Thanks in advance, Tom

**Comment by vovayatsyuk** [comment]
Hi, 

this case (extending custom components) is not supported in the current Breeze version.

I'm working on improving components to use `.extend` in any custom component, just like in your example.

I will let you know when it's ready.

**Reply by nul77** [reply]
Thank you for your reply. I'll keep an eye out for your updates page.

Tom

**Reply by vovayatsyuk** [reply]
Can you pull the latest changes and try it out?

The fix itself: https://github.com/breezefront/module-breeze/commit/7eadf8174869b469beafdc5538b258231d745919

**Reply by nul77** [reply]
Thank yo for your quick reply. Installing the latest version unfortunately results in an error in console when used with swissup/module-breeze-smile-elasticsuite:1.0.3
![console](https://github.com/breezefront/community/assets/11329417/efe6d37c-f67e-463b-bde2-853c44f0f268)


Tom

**Reply by vovayatsyuk** [reply]
Looks like a conflict with the dynamic js feature. I didn't test the integrations yet. Will work on them on Monday.

What about Amasty_Gdpr?

**Reply by nul77** [reply]
Contacted Amasty last week. Haven't heard back.

Tom

**Reply by vikasgamit** [reply]
Have you received any reply from the Amasty team? They have provided an update.

We are stuck on events binds. modal is open, but events are not bound with the modal buttons.

**Reply by nul77** [reply]
Hi,

Amasty said they didn't have any plans to make the module Breeze compatible.
My client opted for another cookiewall solution in the end.

Tom

**Reply by vikasgamit** [reply]
Thank you for your response, @nul77 . I have also generated a support ticket with them; let me see what they reply.
